### ISSUE TITLE
FR: Set annotated image with auto value width

### ISSUE THREAD

**Issue by zanodor** [zsviczian/obsidian-excalidraw-plugin#1678]
Hi Zsolt,

First of all, great plugin. A compatriot of mine leading most downloaded plugin...well, no surprises there... Maybe the language of Ady has something to do with it...

## Problem

I've seen an important-looking use case with the annotation feature. A relevant thread on the Obs. Forum is [here](https://forum.obsidian.md/t/excalidraw-image-annotate-new-feature-templater/79538/).

I've seen a workaround presented by you...but I find it a bit fiddly or messy. Or to put it simply: I lack the expertise to do it. It involves reading of the image file for the width property. I think Excalidraw -- or at least the image created with the annotate command -- could follow Obsidian's default handling of image embeds and not enforce a set width. 

## Solution

How about treating the annotated image with auto width? All excalidraw extension files could still be used with the width value set in the settings. I wouldn't want changes involving all settings and readme.md, etc. 

## Alternative solution

I cannot think of one.

Üdv.
Z.

**Comment by zsviczian** [comment]
You can create multiple Excalidraw templates. To keep it simple I would create 3: template-small, template-med, template-large.
Depending on how you annotate images, you may also want to add a smaller padding for these exports.

Create your css snippet based on your cssclasses
```css
.img-small { width:  300px; }
.img-med { width:  700px; }
.img-large { width:  1000px; }
.img-xl { width: 100% }
```


Based on the image you annotate you can chose which template to apply.

You'll 
```markdown
---
excalidraw-plugin: parsed
excalidraw-export-padding: 0
cssclasses
  - img-w300
---

# Text Elements
....
```

if you want to perscribe size based on css classes you will want to set image width to 100% in export settings.

![image](https://github.com/zsviczian/obsidian-excalidraw-plugin/assets/14358394/3bd7232d-ebcd-4f7f-8db4-4d941ee35bc7)

**Comment by zsviczian** [comment]
enabling Auto is an interesting idea, If you manually set it in developer console do you get the desired result?

**Comment by zsviczian** [comment]
I did not mean editing data.json, but rather editing the HTML in developer console. I tested it now, setting width:auto does not have the desired effect.

The issue is there are already too many features packed on top of one anther regarding width and height not to mention the other features... so dumming down is not really an option. If I implement this it will just add to the complexity, with additional settings and potentially conflicting needs. People have very different needs...

I don't have a better solution for you now, than the one proposed above, and thinking about this, I also don't see this as a feature I would develop further, given the complexity it would add and my fear that if I'd ask 10 people how they wanted this, they would have 10 different preference.
